web/modules/custom/mymodule/src/Form/SimpleForm.php

```diff
@@ -0,0 +1,199 @@
+    return new static(
+      $container->get('messenger')
+    );
+
+  }
+
+  /**
+   * Construct function to call messenger.
+   *
+   * @param \Drupal\Core\Messenger\MessengerInterface $messenger
+   *   Call the messenger interface.
+   */
+  public function __construct(MessengerInterface $messenger) {
+    $this->messenger = $messenger;
+  }
+
+  /**
+   * Get the formid function.
+   *
+   * @return string
+   *   It return the formid.
+   */
+  public function getFormId() {
+    return 'Simple_Form';
+  }
+
+  /**
+   * Buildform function to create a form fields.
+   *
+   * @param array $form
+   *   It return the form fields as array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   It return the stored values of form.
+   *
+   * @return array
+   *   Form is returned as array.
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+
+    $countries = CountryManager::getStandardList();
+    $country_array[''] = 'select';
+    foreach ($countries as $key => $value) {
+      $country_name = (string) $value;
+      $country_array[$country_name] = $country_name;
+    }
+    $formValues = $form_state->get('storelocations');
+    if (empty($formValues)) {
+      $formValues = $form_state->set('storelocations', 1);
+    }
+
+    $form['country'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Country'),
+      '#options' => $country_array,
+      '#key_type' => 'associative',
+    ];
+    $form['locations'] = [
+      '#type' => 'container',
+      '#tree' => TRUE,
+      '#prefix' => '<div id="locations">',
+      '#suffix' => '</div>',
+    ];
+    if ($formValues) {
+      for ($i = 0; $i < $formValues; $i++) {
+        $form['locations'][$i] = [
+          '#type' => 'fieldset',
+          '#tree' => TRUE,
+          '#prefix' => 'Location' . $i,
+        ];
+
+        $form['locations'][$i]['state'] = [
+          '#title' => t('State'),
+          '#type' => 'textfield',
+        ];
+        $form['locations'][$i]['city'] = [
+          '#title' => t('City'),
+          '#type' => 'textfield',
+        ];
+      }
+    }
+    $form['add_another_location'] = [
+      '#type' => 'submit',
+      '#value' => t('Add A Location'),
+      '#submit' => ['::addOne'],
+      '#href' => '',
+      '#ajax' => [
+        'callback' => '::customAjaxAddLocation',
+        'wrapper' => 'locations',
+      ],
+    ];
+    $form_state->setCached(FALSE);
+    $form['submit'] = [
+      '#value' => t('Save'),
+      '#type' => 'submit',
+    ];
+
+    return $form;
+
+  }
+
+  /**
+   * Custom function to call the location fieldset.
+   */
+  public function customAjaxAddLocation(array &$form, FormStateInterface $form_state) {
+    return $form['locations'];
+  }
+
+  /**
+   * Custom function to increment the value of fieldset and stored in formstate.
+   */
+  public function addOne(array &$form, FormStateInterface $form_state) {
+    $formValues = $form_state->get('storelocations');
```

Comment:
The correct name of the variable would be - `$num_locations`. Also, we should only follow one pattern for naming variables - it could either be camelcase or all lowercase. In PHP preferred is `lowercase`

Reply:
Fixed